test/cvs-scm/run.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash -e
+#
+#  Quick functionality test for CVS
+#
+. ../test-lib.sh 2>/dev/null || { echo "Must run in script directory!" ; exit 1 ; }
+
+dir=$(mktemp -d)
+echo "Using scratch dir: $dir"
+
+# Bob recipes
+bob=$dir/bob
+mkdir -p "$bob/recipes"
+cat >"$bob/recipes/t.yaml" <<EOF
+root: true
+checkoutSCM:
+  scm: cvs
+  cvsroot: \${CVSROOT}
+  module: mod
+buildScript: |
+  cp "\$1/file.txt" .
+packageScript: |
+  cp "\$1/file.txt" .
+EOF
```

Comment:
Why not just put the recipes directly in the test directory like the other black box tests?

Reply:
Because I'm updating the recipes during the test to check the "change the branch" usecase. Creating the recipes from scratch was the simplest, safest I came up with.

Reply:
Normally I would say you could use variable substitution and `-D...` when invoking bob. But it looks like it really makes a difference if the `rev` property on the SCM exists or not. :thinking: So maybe it's really the best option.

Reply:
I wanted as little logic as possible in the test, to exercise more paths in the code. For CVS, a

     rev: ${REV}

would have been possible, passing "HEAD" for "not-from-a-branch".

I have now placed the recipes as separate files to save us a layer of quoting, but they are still copied into a fresh workspace.